src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -585,22 +596,60 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
                                 (String)sObj.get('CurrencyIsoCode'),
                                 allo.Amount__c
                             );
+
+                            if (alloInMap != null) {
+                                alloInMap.Amount__c += allo.Amount__c;
+                            }
                         }
                         allo.put('CurrencyIsoCode', sObj.get('CurrencyIsoCode'));
                     }
 
                     if (allo.Percent__c != null) {
                         allo.Amount__c = (amount * allo.Percent__c * .01).setScale(2);
+                        
+                        if (alloInMap != null) {
+                            alloInMap.Amount__c += allo.Amount__c;
```

Comment:
The way you have this and the one on 601  laid out, you will inflate the GAU amount (more than it should be) if it is a multicurrency org, because both sections will be entered.  Should account for the one on 621, as well, to make sure it does not duplicate in either case.  Btw, you need to make sure the isMultiCurrency stuff carries over to your Consolidated GAUs as well.